docs/guide/callbacks.rst

```diff
@@ -0,0 +1,328 @@
+            # local and global variables
+            # self.locals = None  # type: Dict[str, Any]
+            # self.globals = None  # type: Dict[str, Any]
+            # The logger object, used to report things in the terminal
+            # self.logger = None  # type: logger.Logger
+            # # Sometimes, for event callback, it is useful
+            # # to have access to the parent object
+            # self.parent = None  # type: Optional[BaseCallback]
+
+        def _on_training_start(self) -> None:
+            """
+            This method is called before the first rollout starts.
+            """
+            pass
+
+        def _on_rollout_start(self) -> None:
+            """
+            A rollout is the collection of environment interaction
+            using the current policy.
+            This event is triggered before collecting new samples.
+            """
+            pass
+
+        def _on_step(self) -> bool:
+            """
+            This method will be called by the model.
+            This is the equivalent to the callback function.
+            `locals()` and `globals()` are directly accessible as attributes.
+
+            :return: (bool) Return True for continuing training, False for stopping.
+            """
+            return True
+
+        def _on_rollout_end(self) -> None:
+            """
+            This event is triggered before updating the policy.
+            """
+            pass
+
+        def _on_training_end(self) -> None:
+            """
+            This event is triggered before exiting the `learn()` method.
+            """
+            pass
+
+
+.. note::
+  `self.num_timesteps` corresponds to the total number of steps taken in the environment, i.e., it is the number of environments multiplied by the number of time `env.step()` was called
+
+  You should know that `PPO1` and `TRPO` update `self.num_timesteps` after each rollout (and not each step) because they rely on MPI.
+
+  For the other algorithms, `self.num_timesteps` is incremented by `n_envs` (number of environments) after each call to `env.step()`
+
+
+.. note::
+
+  For off-policy algorithms like SAC, DDPG, TD3 or DQN, the notion of `rollout` corresponds to the steps taken in the environment between two updates.
+
+
+
+Event Callback
+--------------
+
+Compared to Keras, Stable Baselines provides a second type of `BaseCallback`, named `EventCallback` that is meant to trigger events. When an event is triggered, then a child callback is called.
+
+As an example, `EvalCallback` is an `EventCallback` that will trigger its child callback when there is a new best model.
+A child callback is for instance `StopTrainingOnRewardThreshold` that stops the training if the mean reward achieved by the RL model is above a threshold.
+
+.. note::
+
+	We recommend to take a look at the source code of `EvalCallback` and `StopTrainingOnRewardThreshold` to have a better overview of what can be achieved with this kind of callbacks.
```

Comment:
Can we have links to those functions so people can quickly jump in?

Reply:
good point